### ISSUE TITLE
[Potential issue] Does it matter that the probabilities are not being renormalized before Temperature is called?

### ISSUE THREAD

**Issue by kalomaze** [ggerganov/llama.cpp#3971]
When running min p (the final sampler), we are not calling `llama_sample_softmax(ctx, candidates);` after truncating the logits. Does this have a discernable impact on how temperature scales afterwards? Or is it just good practice? I'm worried that temperature is not scaling in a way that is linear/non-skewed because of this

**Comment by kalomaze** [comment]
It appears to make no difference in the end calculations, but it might be good for consistency's sake
